pytest/tests/loadtest/locust/common/base.py

```diff
@@ -293,14 +305,55 @@ def evaluate_rpc_result(rpc_result):
 # called once per process before user initialization
 @events.init.add_listener
 def on_locust_init(environment, **kwargs):
-    funding_key = key.Key.from_json_file(environment.parsed_options.funding_key)
-    NearUser.funding_account = Account(funding_key)
+    # Note: These setup requests are not tracked by locust because we use our own http session
+    host, port = environment.host.split(":")
+    node = cluster.RpcNode(host, port)
 
+    master_funding_key = key.Key.from_json_file(
+        environment.parsed_options.funding_key)
+    master_funding_account = Account(master_funding_key)
 
-# CLI args
+    funding_account = None
+    # every worker needs a funding account to create its users, eagerly create them in the master
+    if isinstance(environment.runner, runners.MasterRunner):
```

Comment:
A modern (Python >=3.10) way to match the class of the object would be to use pattern matching [1]:

```python
match environment.runner:
  case runners.MasterRunner:
    # Handle MasterRunner case.
  case runners.WorkerRunner:
    # Handle WorkerRunner case.
  ...
  case other_runner:
    # Handle all other cases.
```


[1] https://peps.python.org/pep-0636/#adding-a-ui-matching-objects

Reply:
Yeah that looks much nicer. 

I don't know our official policy regarding python versions but I try to be on the conservative side to ensure compatibility with preinstalled python on older OSs and docker images. But I could be convinced otherwise quite easily.

Reply:
I would probably optimize for developer productivity here, as we expect this code to be mainly used by devs on machines that they control. In this case, the version should be easy to install on common OSes.
Ubuntu 22.04 is on Python 3.10, Archlinux is on Python 3.11, so 3.10 sounds like a safe bet.

Our Docker setup uses Ubuntu 18.04, but its support is ending in June 2023 [1] and we'll need to migrate to new version that will be >= 20.04 soon.

[1] https://wiki.ubuntu.com/Releases

Reply:
Good points. Well, I know I am still running private Ubuntu 20.04 systems ([python 3.8 by default](https://wiki.ubuntu.com/FocalFossa/ReleaseNotes#Python3_by_default)) but I just noticed the official support for it was dropped a month ago.^^

Anyway, I opened https://github.com/near/nearcore/pull/9125